src/fides/api/ops/api/v1/endpoints/connection_endpoints.py

```diff
@@ -202,34 +211,95 @@ def get_connection_detail(
 def patch_connections(
     *,
     db: Session = Depends(deps.get_db),
-    configs: conlist(CreateConnectionConfiguration, max_items=50),  # type: ignore
+    configs: conlist(CreateConnectionConfigurationWithSecrets, max_items=50),  # type: ignore
 ) -> BulkPutConnectionConfiguration:
     """
-    Given a list of connection config data elements, create or update corresponding ConnectionConfig objects
-    or report failure
+    Given a list of connection config data elements, optionally containing the secrets,
+    create or update corresponding ConnectionConfig objects or report failure
 
     If the key in the payload exists, it will be used to update an existing ConnectionConfiguration.
     Otherwise, a new ConnectionConfiguration will be created for you.
-
-    Note that ConnectionConfiguration.secrets are not updated through this endpoint.
     """
-    created_or_updated: List[ConnectionConfig] = []
+    created_or_updated: List[ConnectionConfigurationWithSecretsResponse] = []
     failed: List[BulkUpdateFailed] = []
     logger.info("Starting bulk upsert for {} connection configuration(s)", len(configs))
 
     for config in configs:
+        if config.connection_type == "saas":
+            if config.secrets:
+                try:
+                    connection_config_check = get_connection_config_or_error(
```

Comment:
I agree this feels clunky, but the alternative also felt clunky. `get_connection_config_or_error` would need to catch the other exception type and convert it to `HTTPException`. This clunky option resulted in less code so I went this direction, but I'm happy to update the other way if you think it would be clearer.

Reply:
makes sense. an alternative i just thought of - since `get_connection_config_or_error` is pretty simple, you could just not use the function at all here and instead simply 
```
connection_config_check = ConnectionConfig.get_by(db, field="key", value=connection_key)
```
and then leave it at that, since if the config doesn't exist, it will be `None` anyway, which is where you end up now too?

Reply:
Yes, I think this makes sense. I'll do that.